types.d.ts

```diff
@@ -83,66 +85,262 @@ type WriteOptions = {
   jsonIndent?: number;
 };
 
+
+// API has the same set of synchronous and asynchronous methods. 
+// All async methods are promise based (no callbacks).
+
+// Commonly used naming convention in Node world is reversed in this library (no 'method' and 'methodSync' naming). 
+// Asynchronous methods are those with 'Async' suffix, all methods without 'Async' in the name are synchronous. 
+// Reason behind this is that it gives very nice look to blocking API. And promise-based, non-blocking code is verbose 
+// anyway, so one more word is not much of a difference.
+
+
 export interface FSJetpack {
+  /**
+   * Returns Current Working Directory (CWD) for this instance of jetpack, or creates new jetpack object with given path as its internal CWD.
+   * 
+   * **Note:** fs-jetpack never changes value of `process.cwd()`, the CWD we are talking about here is internal value inside every jetpack instance.
+   * 
+   *  @param path (optional) path (or many path parts) to become new CWD. Could be absolute, or relative. If relative path given new CWD will be resolved basing on current CWD of this jetpack instance.
+   */
   cwd: {
     (): string;
     (...pathParts: string[]): FSJetpack;
   };
 
   path(...pathParts: string[]): string;
 
+  /**
+   * Appends given data to the end of file. If file or any parent directory doesn't exist it will be created.
+   * 
+   * @param path  the path to file.  
+   * @param data  data to append (can be `String` or `Buffer`).  
+   * @param options 
+   */
   append(path: string, data: AppendData, options?: AppendOptions): void;
+  
+  /**
+   * Appends given data to the end of file. If file or any parent directory doesn't exist it will be created.
+   * 
+   * @param path  the path to file.  
+   * @param data  data to append (can be `String` or `Buffer`).  
+   * @param options 
```

Comment:
How would I know what `options.mode` does and what it defaults to? The docs in readme says:

> options (optional) Object with possible fields:
> 
> mode if the file doesn't exist yet, will be created with given mode. Value could be number (eg. 0o700) or string (eg. '700').

Reply:
I left all the options docs blank, those docs can go in the interfaces for each option, happy to let someone else add those 👍